src/controllers/connection/newConnection.ts

```diff
@@ -413,48 +413,66 @@ export class NewConnectionController extends HTMLController {
   }
 
   private async updateExistingConnection(
+    logger: pino.Logger,
     company: OrganisationProfile,
     pinHash: string,
     invitationId: string
   ): Promise<{ type: 'success'; connectionId: string } | { type: 'error'; error: string }> {
     try {
       // logic in allowNewInvitation ensures only one entry exists for connectionRecord
       const [connectionRecord] = await this.db.get('connection', { company_number: company.company_number })
-      const connectionId = connectionRecord.id // to return the connectionId
-      // reset pin count and invite status to 'pending'
-      await this.db.withTransaction(async (db) => {
-        await db.update(
-          'connection',
-          { company_number: company.company_number },
-          {
-            status: 'pending',
-            pin_attempt_count: 0,
-            pin_tries_remaining_count: null,
+      // to return the connectionId
+      const connectionId = connectionRecord.id
+      // get invitations array
+      // NB we haven't inserted the new invitation into the ui db yet so its OOB invite won't be deleted
+      const invitations = await this.db.get('connection_invite', { connection_id: connectionId })
+      // delete all existing OOB invitations from the cloudagent
+      for (const invite of invitations) {
+        const exists = await this.cloudagent.getOutOfBandInvite(invite.oob_invite_id).catch((error) => {
+          if (error instanceof Error && error.message.endsWith('Not Found')) {
+            logger.debug('OOB invitation already deleted %s', invite.oob_invite_id)
           }
-        )
+        })
```

Comment:
normally it would be a check for a `404` status code rather than checking the error message, is that possible? It looks like this is catching all errors too so any other errors will silently fail

Reply:
Can you help me with this one?